src/jit/gentree.h

```diff
@@ -2199,6 +2198,92 @@ struct GenTree
     inline GenTree(genTreeOps oper, var_types type DEBUGARG(bool largeNode = false));
 };
 
+struct GenTreePhi : public GenTree
```

Comment:
This needs a header with a general description of the node, as well as the constraints around how these are used, e.g.:
- Each of the `Use`s are `GT_LCL_VAR` nodes
- This node is always the rhs of an assignment, the lhs of which is also a `GT_LCL_VAR` node.

Reply:
While I'm at it, I'll mention that I wanted to make `op` `GenTreePhiArg*` rather than `GenTree*`, to make it clear what node types are supported. Unfortunately there are many places that take the address of `op` and pass it around as `GenTree**` so this is impossible to do.